lms/djangoapps/course_home_api/progress/v1/views.py

```diff
@@ -122,11 +129,32 @@ def get(self, request, *args, **kwargs):
 
         enrollment_mode, _ = CourseEnrollment.enrollment_mode_for_user(request.user, course_key)
 
-        course_grade = CourseGradeFactory().read(request.user, course)\
+        # The block structure is used for both the course_grade and has_scheduled content fields
+        # So it is called upfront and reused for optimization purposes
+        collected_block_structure = get_block_structure_manager(course_key).get_collected()
+        course_grade = CourseGradeFactory().read(request.user, collected_block_structure=collected_block_structure)
+
+        # Get has_scheduled_content data
+        transformers = BlockStructureTransformers()
+        transformers += [start_date.StartDateTransformer()]
+        usage_key = collected_block_structure.root_block_usage_key
+        course_blocks = get_course_blocks(
+            request.user,
+            usage_key,
+            transformers=transformers,
+            collected_block_structure=collected_block_structure,
+            include_has_scheduled_content=True
+        )
+        has_scheduled_content = course_blocks.get_xblock_field(usage_key, 'has_scheduled_content')
+
+        # Get user_has_passing_grade data
+        user_has_passing_grade = False
+        if not request.user.is_anonymous:
+            user_grade = course_grade.percent
```

Comment:
Just to confirm, this is a weighed grade, correct? It won't show up as 100% after unit 1, despite there being no way they have passed the full course.

Reply:
i just copied the code when moving user_has_passing_grade out from the course home api per your suggestion, so i think it makes sense to keep it consistent with the existing variable
not really sure about the course grade code but i didn't see that behavior locally / this variable seems to work fine on course end

Reply:
Ah I see, okay. For future reference purposes, I think you mean courseware_api, but I get what you mean.

Reply:
i meant that i previously had this variable it in the course home api in this PR and you suggested moving it to the progress api, but yeah i'm copying it from the courseware api